tests/schema/test_union.py

```diff
@@ -968,3 +968,66 @@ def some_type_queries(self, id: strawberry.ID) -> ObjectQueries[SomeType]:
             """
         ).strip()
     )
+
+
+def test_annoted_union_with_two_generics():
+    @strawberry.type
+    class SomeType:
+        a: str
+
+    @strawberry.type
+    class OtherType:
+        b: str
+
+    @strawberry.type
+    class NotFoundError:
+        message: str
+
+    T = TypeVar("T")
+    U = TypeVar("U")
+
+    @strawberry.type
+    class UnionObjectQueries(Generic[T, U]):
+        @strawberry.field
+        def by_id(
+            self, id: strawberry.ID
+        ) -> T | Annotated[U | NotFoundError, strawberry.union("ByIdResult")]: ...
```

Comment:
This test captures the combined case of one generic inside and one generic outside the annotation. Both in and both out also work, but I figured the tests would be getting a bit redundant if I included every combination of in and out.